src/runtime/ThreadLocal.cpp

```diff
@@ -131,6 +142,45 @@ void ThreadLocal::initialize()
     // Heap is initialized for each thread
     Heap::initialize();
 
+    // g_stackLimit
+#if defined(OS_WINDOWS)
+    ULONG_PTR low, high;
+    GetCurrentThreadStackLimits(&low, &high);
+    void* stackStartAddress = reinterpret_cast<void*>(low);
+    void* stackEndAddress = reinterpret_cast<void*>(high);
+    size_t stackSize = reinterpret_cast<size_t>(stackEndAddress) - reinterpret_cast<size_t>(stackStartAddress);
+#else
+    void* stackStartAddress;
+    void* stackEndAddress;
+    size_t stackSize;
+#if defined(OS_DARWIN)
+    stackStartAddress = pthread_get_stackaddr_np(pthread_self());
+    stackSize = pthread_get_stacksize_np(pthread_self());
+#else
+    pthread_attr_t attr;
+    pthread_getattr_np(pthread_self(), &attr);
+    pthread_attr_getstack(&attr, &stackStartAddress, &stackSize);
+    pthread_attr_destroy(&attr);
+#endif
+#ifdef STACK_GROWS_DOWN
+    stackEndAddress = (char*)stackStartAddress - stackSize;
+#if defined(OS_DARWIN)
+    std::swap(stackStartAddress, stackEndAddress);
+#endif
+#else
+    stackEndAddress = (char*)stackStartAddress + stackSize;
+#endif
+#endif
+    stackSize = std::min(stackSize, (size_t)STACK_USAGE_LIMIT);
```

Comment:
It seems that `stackSize = std::min(stackSize, (size_t)STACK_USAGE_LIMIT);` is not used afterward.

Reply:
Oops my mistake